agent/bzero/registration/registration.go

```diff
@@ -134,13 +126,28 @@ func post(log logger.T, regInfo BZeroRegRequest, regUrl string) (*http.Response,
 			Timeout: time.Second * 10,
 		}
 
+		// Build request
```

Comment:
Is there a reason we moved this code into the for loop? Now we are building the request for every iteration of the for loop which would be less efficient than just building it once and calling `response, err = httpClient.Do(req)`? Or maybe I misunderstood something.

Reply:
yeah, this what was causing the bug so we need to build it anew every time